python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -234,15 +238,48 @@ class TimeWindow(NamedTuple):
 
 @whitelist_for_serdes(
     storage_name="TimeWindow",  # For back-compat with existing serdes
-    field_serializers={"start": DatetimeFieldSerializer, "end": DatetimeFieldSerializer},
 )
-class PersistedTimeWindow(NamedTuple):
+class PersistedTimeWindow(
+    NamedTuple(
+        "_PersistedTimeWindow", [("start", TimestampWithTimezone), ("end", TimestampWithTimezone)]
+    )
+):
     """Internal serialized representation of a time interval that is closed at the
     start and open at the end.
     """
 
-    start: datetime
-    end: datetime
+    def __new__(
+        cls,
+        start: Union[TimestampWithTimezone, PendulumDateTime],
+        end: Union[TimestampWithTimezone, PendulumDateTime],
+    ):
+        return super(cls, PersistedTimeWindow).__new__(
+            cls,
+            start=(
+                start
+                if isinstance(start, TimestampWithTimezone)
+                else TimestampWithTimezone.from_pendulum_time(start)
+            ),
+            end=(
+                end
+                if isinstance(end, TimestampWithTimezone)
+                else TimestampWithTimezone.from_pendulum_time(end)
+            ),
+        )
+
+    @property
```

Comment:
potentially a small perf hit here from no longer pre-computing this (the lack of cached_property on NamedTuples is extremely annoying)

Reply:
are you sure `cached_property` won't work here? we use it in TimeWindowPartitionsSubset: https://sourcegraph.com/github.com/dagster-io/dagster/-/blob/python_modules/dagster/dagster/_core/definitions/time_window_partitions.py?L2215